cellxgene_schema_cli/cellxgene_schema/utils.py

```diff
@@ -202,3 +202,36 @@ def is_ontological_descendant_of(onto: OntologyParser, term: str, target: str, i
 @lru_cache()
 def get_descendants(onto: OntologyParser, term: str, include_self: bool = True) -> List[str]:
     return onto.get_term_descendants(term, include_self=True)
+
+def check_non_csr_matrices(adata: ad.AnnData):
+    """
+    Check X, raw.X and layers matrices for having more than 50% zeros and not being csr_matrix
+
+    If found, convert to csr_matrix
+    """
+
+    def get_sparsity(matrix, format):
+        if format in SPARSE_MATRIX_TYPES:
+            nnz = matrix.nnz
+        else:
+            nnz = np.count_nonzero(matrix)
+        sparsity = 1 - nnz / np.prod(matrix.shape)
+        return sparsity
+
+    format = get_matrix_format(adata, adata.X)
+    if format != 'csr' and get_sparsity(adata.X, format) >= .5:
+        adata.X = sparse.csr_matrix(adata.X)
+
+    format = get_matrix_format(adata, adata.raw.X)
```

Comment:
For datasets that do not have a normalized matrix and only have raw matrix, there not be a matrix in adata.raw.X. So, I believe this needs to be under an if-statement.